fetch/fetch.go

```diff
@@ -473,12 +545,12 @@ func (f *Fetch) sendBatch(p p2p.Peer, requests []RequestMessage) error {
 			return nil
 		}
 
-		f.log.With().Debug("sending request batch to peer",
+		f.logger.With().Debug("sending request batch to peer",
 			log.String("batch_hash", batch.ID.ShortString()),
 			log.Int("num_requests", len(batch.Requests)),
 			log.String("peer", p.String()))
 
-		err = f.hashSrv.Request(context.TODO(), p, bytes, f.receiveResponse, errorFunc)
+		err = f.servers[hashProtocol].Request(context.TODO(), p, bytes, f.receiveResponse, errorFunc)
```

Comment:
`context.TODO()`! Could that be the context that is canceled when `Stop()` is called?

Reply:
yes. i haven't got around to change this. it will return after timeout even if app is shutting down. added an app context to fetch.